### PR TITLE
Fix #45: support testing multiple commits in one PR

### LINKED ISSUE #45: support bench specified commits of a PR
support bench specified commits of a PR

### PR DESCRIPTION
**Pull request by liufengyun**
Fix #45: support testing multiple commits in one PR

### CODE REVIEW

**Comment by liufengyun** [comment]
Review @smarter?

**Review by smarter — APPROVED** [review]
LGTM! A possible improvement would be to support commit ranges like `foo..bar`. You can use `git rev-list foo..bar` to get the list of commits between a start and an end commit.

**Comment by liufengyun** [comment]
Thanks @smarter . We can do range when current command becomes annoying :)

File: bin/pull [hunk 1]
```diff
@@ -68,27 +86,36 @@ cp history.csv $FILE
...
+############ test commits
+
+for origin_commit in $(echo $COMMITS | tr ":" "\n")
+do
+  # do a local merge
+  commit_time=$(date -u +"%Y-%m-%dT%H:%M:%SZ")   # use current time so the point appear last
+
+  cd $PROG_HOME/dotty
+  git reset --hard origin/master
+  git merge $origin_commit -m "merge PR #$PR commit $origin_commit with master" > $LOG 2>&1
```
Comment (smarter) [inline bin/pull:113]: shouldn't `>>` be used instead of `>` to not overwrite the log every time?

Comment (liufengyun) [reply bin/pull:113]: Good catch, I'm pushing a fix soon.

File: bin/poll [hunk 2]
```diff
@@ -68,15 +74,27 @@ while true; do
...
 
-  pulls=$(cat $output | jq -r '.[] | select((.body | contains("test performance please")) and (.html_url | contains("pull")) )  | .html_url')
+  pulls=$(cat $output | jq -r '.[] | select((.body | contains("test performance please")) and (.html_url | contains("pull")) ) | [.html_url,  .body] | @tsv')
 
   while IFS= read -r line; do
     if [ "$line" != "" ]; then
       # https://github.com/lampepfl/dotty/pull/2970#issuecomment-321860528
       pr=$(echo $line | sed 's/^.*pull\/\([0-9]\+\)#.*/\1/')
+
+      commits_info=$(echo $line | sed 's/.*test performance please *:\(.*\)$/\1/' | sed 's/[^1-9a-z ]//g') # sanitize input
```
Comment (smarter) [inline bin/poll:84]: This sould be 0-9, not 1-9

**Pull request merged**